pkg/ibm/resources_container.go

```diff
@@ -60,20 +61,20 @@ func (tagsCollector *tagsClient) setResourceTags(resource datamodel.TaggedResour
 // ResourcesContainer holds the results of collecting the configurations of all resources.
 type ResourcesContainer struct {
 	datamodel.ResourcesContainerModel
-	regions         []string
-	ResourceGroupID string
+	regions               []string
+	ResourceGroupIDOrName string
```

Comment:
I suggest to leave resourceGroupId as the class member like we had before. It looks strange to pass a parameter to other methods, when you have this member.
Then, maybe just rename `getResourceGroupID()` to `verifyResourceGroupID()` and call it from `CollectResourcesFromAPI()`.